opentelemetry-sdk/src/opentelemetry/sdk/_metrics/_internal/export/__init__.py

```diff
@@ -193,23 +205,28 @@ def _at_fork_reinit(self):
     def _ticker(self) -> None:
         interval_secs = self._export_interval_millis / 1e3
         while not self._shutdown_event.wait(interval_secs):
-            self.collect()
+            self.collect(timeout_millis=self._export_timeout_millis)
         # one last collection below before shutting down completely
-        self.collect()
+        self.collect(timeout_millis=self._export_interval_millis)
 
     def _receive_metrics(
-        self, metrics: Iterable[Metric], *args, **kwargs
+        self,
+        metrics: Iterable[Metric],
+        timeout_millis: float = 10_000,
+        **kwargs,
     ) -> None:
         if metrics is None:
             return
         token = attach(set_value(_SUPPRESS_INSTRUMENTATION_KEY, True))
         try:
-            self._exporter.export(metrics)
+            self._exporter.export(metrics, timeout_millis=timeout_millis)
         except Exception as e:  # pylint: disable=broad-except,invalid-name
             _logger.exception("Exception while exporting metrics %s", str(e))
         detach(token)
 
-    def shutdown(self, *args, **kwargs):
+    def shutdown(self, timeout_millis: float = 30_000, **kwargs) -> None:
```

Comment:
Shouldn't there be a configurable `shutdown_timeout_millis` as part of `metricreader` instantiation? This would follow the similar pattern of configuring `export_timeout_millis`.

Reply:
Are you talking about the default value? I guess we could do that for the autoinstrumentation case or the atexit handler. But we would need to make the default here None and defer to the `shutdown_timeout_millis` option.

Reply:
Not a blocker for this PR. Was just wondering if want shutdown to be configurable as well.